plugin/cassandra/plugin.go

```diff
@@ -90,106 +90,151 @@ package main
 		Features: plugin.PluginFeatures{
 			Target: "yes",
 			Store:  "no",
 		},
 		Example: `
 {
-  "host"         : "127.0.0.1",      # optional
-  "port"         : "9042",           # optional
-  "user"         : "username",
-  "password"     : "password",
-  "keyspace"     : "db",
-  "datadir"      : "/path/to/data"   # optional
+  "host"              : "127.0.0.1",      # optional
+  "port"              : "9042",           # optional
+  "user"              : "username",
+  "password"          : "password",
+  "include_keyspaces" : ["db"],
+  "exclude_keyspaces" : ["system"],
+  "save_users"        : true,
+  "bindir"            : "/path/to/bin",   # optional
+  "datadir"           : "/path/to/data",  # optional
 }
 `,
 		Defaults: `
 {
-  "host"     : "127.0.0.1",
-  "port"     : "9042",
-  "user"     : "cassandra",
-  "password" : "cassandra",
-  "datadir"  : "/var/vcap/store/cassandra/data"
+  "host"              : "cassandra_host", // "12.0.0.1"
+  "port"              : "9042", // "cassandra_port"
+  "user"              : "cassandra_user",     //cassandra
+  "password"          : "cassandra_password", // "cassandra"
+  "include_keyspaces" : [""],
+  "exclude_keyspaces" : [ "system_schema", "system_distributed", "system_auth", "system", "system_traces" ],
+  "save_users"        : true,
+  "bindir"            : "/var/vcap/jobs/cassandra/bin", 
+  "datadir"           : "/var/vcap/store/cassandra/data" 
 }
 `,
 		Fields: []plugin.Field{
-			plugin.Field{
+			{
 				Mode:     "target",
 				Name:     "host",
 				Type:     "string",
 				Title:    "Cassandra Host",
 				Help:     "The hostname or IP address of your Cassandra server.",
 				Default:  "127.0.0.1",
-				Required: true,
+				Required: false,
 			},
-			plugin.Field{
+			{
 				Mode:     "target",
 				Name:     "port",
 				Type:     "port",
 				Title:    "Cassandra Port",
 				Help:     "The 'native transport' TCP port that Cassandra server is bound to, listening for incoming connections.",
 				Default:  "9042",
-				Required: true,
+				Required: false,
 			},
-			plugin.Field{
+			{
 				Mode:     "target",
 				Name:     "user",
 				Type:     "string",
 				Title:    "Cassandra Username",
 				Help:     "Username to authenticate to Cassandra as.",
 				Default:  "cassandra",
-				Required: true,
+				Required: false,
 			},
-			plugin.Field{
+			{
 				Mode:     "target",
-				Name:     "password",
+				Name:     "cassandra",
 				Type:     "password",
 				Title:    "Cassandra Password",
-				Help:     "Password to authenticate to Cassandra as.",
+				Help:     "Password to auth_userenv",
 				Default:  "cassandra",
-				Required: true,
+				Required: false,
 			},
-			plugin.Field{
+			{
 				Mode:     "target",
-				Name:     "keyspace",
-				Type:     "string",
-				Title:    "Keyspace to Backup",
-				Help:     "The name of the keyspace to backup.",
-				Example:  "system",
-				Required: true,
+				Name:     "include_keyspaces",
+				Type:     "array",
```

Comment:
Does this work in the SHIELD v8 UI?  I am uncertain that the UI can handle multi-valued fields and turn them into the proper JSON.  May need to revisit how keyspace is specified.